torch/ao/nn/quantized/modules/batchnorm.py

```diff
@@ -48,6 +48,47 @@ def from_reference(cls, bn, output_scale, output_zero_point):
         return qbn
 
 
+class BatchNorm1d(_BatchNorm):
+    r"""This is the quantized version of :class:`~torch.nn.BatchNorm1d`."""
+
+    _NNI_BN_RELU_MODULE = nni.BNReLU2d
+
+    def __init__(
+        self, num_features, eps=1e-5, momentum=0.1, device=None, dtype=None
+    ) -> None:
+        factory_kwargs = {"device": device, "dtype": dtype}
+        super().__init__(num_features, eps, momentum, **factory_kwargs)
+
+    def _get_name(self):
+        return "QuantizedBatchNorm1d"
+
+    def _check_input_dim(self, input):
+        # Temporarily using len(shape) instead of ndim due to JIT issue
+        # https://github.com/pytorch/pytorch/issues/23890
+        if len(input.shape) not in {2, 3}:
+            raise ValueError("Input shape must be `(N, C)` or `(N, C, L)`!")
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        # disabling this since this is not symbolically traceable
+        # self._check_input_dim(input)
+        return torch.ops.quantized.batch_norm1d(
+            input,
+            self.weight,
+            self.bias,
+            self.running_mean,
+            self.running_var,
+            self.eps,
+            self.scale,
+            self.zero_point,
+        )
+
+    @classmethod
+    def from_float(cls, mod, use_precomputed_fake_quant=False):
+        return _BatchNorm.from_float(
+            cls, mod, use_precomputed_fake_quant=use_precomputed_fake_quant
+        )
+
+
 class BatchNorm2d(_BatchNorm):
     r"""This is the quantized version of :class:`~torch.nn.BatchNorm2d`."""
 
```

Comment:
are you enabling this for eager mode quantization? probably add an entry to https://github.com/pytorch/pytorch/blob/ae351d4d0ee0676b81f58170595d016d40cd223f/torch/ao/quantization/quantization_mappings.py#L79 and test the eager mode flow for bn1d as well

Reply:
Thanks, I've added the mapping. Where might the test go? In https://github.com/pytorch/pytorch/blob/main/test/quantization/eager/test_quantize_eager_qat.py?

Reply:
@jerryzh168 is there anything more I should do for this?